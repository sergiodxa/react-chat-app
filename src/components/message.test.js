import React from "react";
import { render } from "@testing-library/react";

import Message from "./message";

test("Message component active", () => {
  const date = new Date("2019-05-03T00:45:43.963Z");
  const { asFragment } = render(
    <Message
      username="test-user"
      date={date}
      content="This is a test"
      now={date.getTime() - 1}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});

test("Message component non active", () => {
  const date = new Date("2019-05-03T00:45:43.963Z");
  const { asFragment } = render(
    <Message
      username="test-user"
      date={date}
      content="This is a test"
      now={date.getTime() + 1}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
