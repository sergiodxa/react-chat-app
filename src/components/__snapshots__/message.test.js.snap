// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component active 1`] = `
<DocumentFragment>
  .emotion-3 {
  padding: 1rem 1.5rem;
  border-bottom: 1px solid #eaeaea;
}

.emotion-3:last-child {
  border-bottom: none;
}

.emotion-0 {
  font-size: 1.1rem;
}

.emotion-1 {
  font-size: .75rem;
  font-style: normal;
  color: #666;
}

.emotion-2 {
  margin: .5rem 0 0;
}

<article
    class="emotion-3"
  >
    <div>
      <strong
        class="emotion-0"
      >
        test-user
      </strong>
       
      <time
        class="emotion-1"
      >
        19:45:43
      </time>
    </div>
    <p
      class="emotion-2"
    >
      This is a test
    </p>
  </article>
</DocumentFragment>
`;

exports[`Message component non active 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 1.1rem;
}

.emotion-1 {
  font-size: .75rem;
  font-style: normal;
  color: #666;
}

.emotion-2 {
  margin: .5rem 0 0;
}

.emotion-3 {
  padding: 1rem 1.5rem;
  border-bottom: 1px solid #eaeaea;
  opacity: .5;
}

.emotion-3:last-child {
  border-bottom: none;
}

<article
    class="emotion-3"
  >
    <div>
      <strong
        class="emotion-0"
      >
        test-user
      </strong>
       
      <time
        class="emotion-1"
      >
        19:45:43
      </time>
    </div>
    <p
      class="emotion-2"
    >
      This is a test
    </p>
  </article>
</DocumentFragment>
`;
